### ISSUE TITLE
Add WebDAV methods

### ISSUE THREAD

**Issue by Henni** [a85/POSTMan-Chrome-Extension#1710]
As @mxk1235 mentioned at https://github.com/postmanlabs/postman-app-support/issues/166#issuecomment-165262522, Postman is missing most of the WebDAV http methods, including:
#### [RFC 4918](http://www.webdav.org/specs/rfc4918.html)
- [ ] PROPPATCH
- [ ] MKCOL
- [ ] MOVE
#### [RFC 3253](http://www.webdav.org/specs/rfc3253.html)
- [ ] VERSION-CONTROL
- [ ] REPORT
- [ ] UPDATE
#### [RFC 5323](http://www.webdav.org/specs/rfc5323.html)
- [ ] SEARCH

**Comment by felixfbecker** [comment]
Need these. It doesn't make much sense to provide COPY but not MOVE. Since Postman is based on Electron, which is build on NodeJS, why not use [`require('http').METHODS`](https://nodejs.org/api/http.html#http_http_methods)?

There are also more important methods like NOTIFY, SUBSCRIBE and UNSUBSCRIBE (used in MS Exchange):
https://msdn.microsoft.com/en-us/library/aa142917(v=exchg.65).aspx

The WebDAV version control RFCs also define more methods, like CHECKOUT. A handy overview over the official IANA registered methods and the RFC where they are defined can be found here: http://www.iana.org/assignments/http-methods/http-methods.xhtml

There are also methods like PURGE and BAN that handle caching, supported by Varnish and Squid
https://www.varnish-cache.org/docs/3.0/tutorial/purging.html
http://wiki.squid-cache.org/SquidFaq/OperatingSquid#How_can_I_purge_an_object_from_my_cache.3F

Shouldn't be such a big effort to add these, right? Currently I cannot use Postman to define collection for the APIs I want because they are missing important methods.

As an alternative you could allow to type in a custom string as a method, since HTTP actually defines that the methods can be extended.

``` js
require('http').METHODS
[ 'ACL',
  'BIND',
  'CHECKOUT',
  'CONNECT',
  'COPY',
  'DELETE',
  'GET',
  'HEAD',
  'LINK',
  'LOCK',
  'M-SEARCH',
  'MERGE',
  'MKACTIVITY',
  'MKCALENDAR',
  'MKCOL',
  'MOVE',
  'NOTIFY',
  'OPTIONS',
  'PATCH',
  'POST',
  'PROPFIND',
  'PROPPATCH',
  'PURGE',
  'PUT',
  'REBIND',
  'REPORT',
  'SEARCH',
  'SUBSCRIBE',
  'TRACE',
  'UNBIND',
  'UNLINK',
  'UNLOCK',
  'UNSUBSCRIBE' ]
```

**Comment by felixfbecker** [comment]
## Workaround:

Export your collection, open the JSON in an editor, change the method, import it again (select Replace).
Now you can also simply duplicate the request with the custom method to add more routes or change it from your history.

**Comment by jbasko** [comment]
The workaround doesn't work for PROPPATCH for which I have an XML body. The body tab isn't enabled and isn't sent even though the original request and the exported json file contains body specification. Content-Type header is set to application/xml. Interestingly, when I click on "Code" button an choose to generate say Python requests code, the body is included.

Using Postman 5.0.2

**Comment by dehart** [comment]
Since the first request was made back on 11 Dec 2012 . Should we still wait for this to happen?
I wonder why it isn't a priority to make Postman a great tool for DAV developers..  😞

**Comment by llamafilm** [comment]
@felixfbecker this workaround fails to import when I change the method to MKCOL.

**Comment by sankalp0o** [comment]
Rather then supporting WebDAV specific methods, we're working towards allowing users to send and save custom methods in Postman. It's been late in coming but hopefully this will solve a bunch of use cases different people have.

![method list - option 2](https://user-images.githubusercontent.com/7601762/43393889-522150ac-9416-11e8-968b-6e83664fac6a.png)

**Comment by aptem-k** [comment]
@llamafilm As a workaround for MKCOL I created HEAD request instead and in pre-request script ran MKCOL with the same URL.

**Comment by a85** [comment]
Everyone: this is going to be out in 6.5 soon releasing over the next 2 weeks!

**Comment by a85** [comment]
This is available in 6.5 now!
